### PR TITLE
BUG: Handle empty data in FREQ location_identifier

### LINKED ISSUE #433: location_identifier raises key error if all staypoints are pre filtered
If the tracking quality is bad and all staypoints are pre-filtered, then the variable `method_val` does not get the column purpose and therefore [the line 74](https://github.com/mie-lab/trackintel/blob/master/trackintel/analysis/location_identification.py#L74) raises a key error. 

I think the behavior should be that the staypoints are returned with a nan-filled or empty purpose column.

### PR DESCRIPTION
**Pull request by bifbof**
closes #433

### CODE REVIEW

File: trackintel/analysis/location_identification.py [hunk 1]
```diff
@@ -197,6 +197,8 @@ def freq_method(staypoints, *labels):
             group["duration"] = group["finished_at"] - group["started_at"]
```
Comment (hongyeehh) [inline trackintel/analysis/location_identification.py:197]: It's not related to this PR, but I guess this line can be moved out of the `for name, group in sp.groupby("user_id"):` for better readability?

File: trackintel/analysis/location_identification.py [hunk 2]
```diff
@@ -197,6 +197,8 @@ def freq_method(staypoints, *labels):
...
         # pandas keeps inner order of groups
         sp.loc[sp["user_id"] == name, "purpose"] = _freq_transform(group, *labels)
+    if "purpose" not in sp.columns:  # if empty sp
+        sp["purpose"] = None
```
Comment (hongyeehh) [inline trackintel/analysis/location_identification.py:201]: Still related to the data type: shall we use `None`, `pd.NA` or `np.nan`? 

Edit: I think we need to use `pd.NA` for a consistent datatype of `Int64`.

**Review by hongyeehh — CHANGES_REQUESTED** [review]

File: trackintel/analysis/location_identification.py (hunk 1 above)
Comment (bifbof) [reply trackintel/analysis/location_identification.py:197]: I think I did it so I wouldn't have to delete it from the `sp` `DataFrame` later, but I'm not sure.

File: trackintel/analysis/location_identification.py (hunk 2 above)
Comment (bifbof) [reply trackintel/analysis/location_identification.py:201]: As the data type of "purpose" is `object` all three should work. But as `pd.NA` is intended for `Int64` and `np.nan` for `float`, I thought that `None` is good for `object`.

File: trackintel/analysis/location_identification.py (hunk 1 above)
Comment (hongyeehh) [reply trackintel/analysis/location_identification.py:197]: good point, I'll resolve this comment

**Pull request merged**